internal/clientconn/conn.go

```diff
@@ -164,6 +164,8 @@ func (c *conn) run(ctx context.Context) (err error) {
 			return
 		}
 
+		reqConext, reqCancel := context.WithCancel(ctx)
```

Comment:
@AlekSi I guess, alternatively, it could be possible to share context for the whole `c.netConn`, but request-based context looks more reasonable to me.